flink-table/flink-table-api-scala/src/main/scala/org/apache/flink/table/api/expressionDsl.scala

```diff
@@ -1284,47 +490,85 @@ trait ImplicitExpressionConversions {
    * @see TableEnvironment#createTemporaryFunction
    * @see TableEnvironment#createTemporarySystemFunction
    */
-  def call(path: String, params: Expression*): Expression = {
-    lookupCall(path, params: _*)
-  }
+  def call(path: String, params: Expression*): Expression = Expressions.call(path, params: _*)
+
+  /**
+   * A call to an unregistered, inline function. For functions that have been registered before and
+   * are identified by a name, use [[call(String, Object...)]].
+   */
+  def call(function: UserDefinedFunction, params: Expression*): Expression = Expressions.call(
+    function,
+    params: _*)
 
   // ----------------------------------------------------------------------------------------------
   // Implicit expressions in prefix notation
   // ----------------------------------------------------------------------------------------------
 
+  /**
+   * Creates a SQL literal.
+   *
+   * The data type is derived from the object's class and its value.
+   *
+   * For example:
+   *
+   *  - `lit(12)`` leads to `INT`
+   *  - `lit("abc")`` leads to `CHAR(3)`
+   *  - `lit(new BigDecimal("123.45"))` leads to `DECIMAL(5, 2)`
```

Comment:
`java.math.BigDecimal` because Scala decimals are not supported. Or we convert them ourselves in this method?